rust/src/model/telemetry.rs

```diff
@@ -61,3 +63,46 @@ pub enum CommandState {
     Completed,
     Error,
 }
+
+#[derive(Debug, Clone)]
+pub struct Location {
+    pub data: String,
+    pub timestamp: NaiveDateTime,
+    pub device_id: String,
+}
+
+impl Serialize for Location {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut s = serializer.serialize_struct("Location", 3)?;
+        s.serialize_field("data", &self.data)?;
+        s.serialize_field("device_id", &self.device_id)?;
+        s.serialize_field("timestamp", &self.timestamp.format(DATE_FORMAT).to_string())?;
+        s.end()
+    }
+}
```

Comment:
Can you explain what this custom Serializer is for, please :) ?

Reply:
Yeah!, the NaiveDateTime type from chrono does not implement Serialize, so I can't automatically derive Serialize to the Location struct.